DataFormats/Scouting/interface/Run3ScoutingVertex.h

```diff
@@ -57,9 +69,12 @@ class Run3ScoutingVertex {
   float x_;
   float y_;
   float z_;
-  float zError_;
   float xError_;
   float yError_;
+  float zError_;
```

Comment:
Even if ROOT's schema evolution should handle the case of member data reordering, it would be a tiny bit safer to keep the earlier order (in the member data, I'm not against in changing the order in constructor or member functions).

Reply:
Will do.

Reply:
> I'm not against in changing the order in constructor or member functions .

Wouldn't it break the tests when reading the old files stored in cms-data ?
The patch below [1] allows to pass the first part of the unit tests, but then reading then reading the file in `cms-data` leads to:

```
The format of a data product has changed.

%MSG
Begin processing the 1st record. Run 1, Event 1, LumiSection 1 on stream 0 at 21-Jan-2024 16:51:02.358 CET
----- Begin Fatal Exception 21-Jan-2024 16:51:02 CET-----------------------
An exception of category 'TestFailure' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'path'
   [2] Calling method for module TestReadRun3Scouting/'testReadRun3Scouting'
Exception Message:
TestReadRun3Scouting::analyze, analyzeVertexes, xError does not equal expected value
----- End Fatal Exception -------------------------------------------------
```

IIUC this happens because the file stored in `cms-data` was written with a different version of `DataFormats/Scouting/test/TestWriteRun3Scouting.cc` in which the order in the constructor was different.

[1]
```diff
diff --git a/DataFormats/Scouting/interface/Run3ScoutingVertex.h b/DataFormats/Scouting/interface/Run3ScoutingVertex.h
index fc436acfb16..7194bca82af 100644
--- a/DataFormats/Scouting/interface/Run3ScoutingVertex.h
+++ b/DataFormats/Scouting/interface/Run3ScoutingVertex.h
@@ -58,7 +58,7 @@ public:
   float yError() const { return yError_; }
   float zError() const { return zError_; }
   float xyCov() const { return xyCov_; }
-  float xzCov() const { return xyCov_; }
+  float xzCov() const { return xzCov_; }
   float yzCov() const { return yzCov_; }
   int tracksSize() const { return tracksSize_; }
   float chi2() const { return chi2_; }
diff --git a/DataFormats/Scouting/test/TestReadRun3Scouting.cc b/DataFormats/Scouting/test/TestReadRun3Scouting.cc
index 5ccd4bab619..d817b04be6c 100644
--- a/DataFormats/Scouting/test/TestReadRun3Scouting.cc
+++ b/DataFormats/Scouting/test/TestReadRun3Scouting.cc
@@ -168,8 +168,8 @@ namespace edmtest {
     if (expectedTrackIntegralValues_.size() != 5) {
       throwWithMessageFromConstructor("test configuration error, expectedTrackIntegralValues must have size 5");
     }
-    if (expectedVertexFloatingPointValues_.size() != 7) {
-      throwWithMessageFromConstructor("test configuration error, expectedVertexFloatingPointValues must have size 7");
+    if (expectedVertexFloatingPointValues_.size() != 10) {
+      throwWithMessageFromConstructor("test configuration error, expectedVertexFloatingPointValues must have size 10");
     }
     if (expectedVertexIntegralValues_.size() != 3) {
       throwWithMessageFromConstructor("test configuration error, expectedVertexIntegralValues must have size 3");
@@ -1138,19 +1138,29 @@ namespace edmtest {
       if (vertex.z() != expectedVertexFloatingPointValues_[2] + offset) {
         throwWithMessage("analyzeVertexes, z does not equal expected value");
       }
-      if (vertex.zError() != expectedVertexFloatingPointValues_[3] + offset) {
-        throwWithMessage("analyzeVertexes, zError does not equal expected value");
-      }
-      if (vertex.xError() != expectedVertexFloatingPointValues_[4] + offset) {
+      if (vertex.xError() != expectedVertexFloatingPointValues_[3] + offset) {
         throwWithMessage("analyzeVertexes, xError does not equal expected value");
       }
-      if (vertex.yError() != expectedVertexFloatingPointValues_[5] + offset) {
+      if (vertex.yError() != expectedVertexFloatingPointValues_[4] + offset) {
         throwWithMessage("analyzeVertexes, yError does not equal expected value");
       }
+      if (vertex.zError() != expectedVertexFloatingPointValues_[5] + offset) {
+        throwWithMessage("analyzeVertexes, zError does not equal expected value");
+      }
+      if (vertex.xyCov() != expectedVertexFloatingPointValues_[6] + offset) {
+        throwWithMessage("analyzeVertexes, xyCov does not equal expected value");
+      }
+      if (vertex.xzCov() != expectedVertexFloatingPointValues_[7] + offset) {
+        std::cout << vertex.xzCov() << "!=" << expectedVertexFloatingPointValues_[7] + offset << std::endl;
+        throwWithMessage("analyzeVertexes, xzCov does not equal expected value");
+      }
+      if (vertex.yzCov() != expectedVertexFloatingPointValues_[8] + offset) {
+        throwWithMessage("analyzeVertexes, yzCov does not equal expected value");
+      }
       if (vertex.tracksSize() != expectedVertexIntegralValues_[0] + iOffset) {
         throwWithMessage("analyzeVertexes, tracksSize does not equal expected value");
       }
-      if (vertex.chi2() != expectedVertexFloatingPointValues_[6] + offset) {
+      if (vertex.chi2() != expectedVertexFloatingPointValues_[9] + offset) {
         throwWithMessage("analyzeVertexes, chi2 does not equal expected value");
       }
       if (vertex.ndof() != expectedVertexIntegralValues_[1] + iOffset) {
diff --git a/DataFormats/Scouting/test/test_readRun3Scouting_cfg.py b/DataFormats/Scouting/test/test_readRun3Scouting_cfg.py
index 01ee20dc64f..a31eef0377f 100644
--- a/DataFormats/Scouting/test/test_readRun3Scouting_cfg.py
+++ b/DataFormats/Scouting/test/test_readRun3Scouting_cfg.py
@@ -97,7 +97,7 @@ process.testReadRun3Scouting = cms.EDAnalyzer("TestReadRun3Scouting",
     tracksTag = cms.InputTag("run3ScoutingProducer", "", "PROD"),
     expectedVertexFloatingPointValues = cms.vdouble(
         15.0,   25.0,  35.0,  45.0,  55.0,
-        65.0,   75.0
+        65.0,   75.0,  85.0,  95.0,  105.0
     ),
     expectedVertexIntegralValues = cms.vint32(
         15,  25,  35
```

Reply:
@mmasciov please take this into account for the next push.

Reply:
Done in [559a119](https://github.com/cms-sw/cmssw/pull/43758/commits/559a119fff075624ae95da320e667e015feb06a8): I left the previous order untouched to minimize potential issues, and updated a couple of files in `test` that I had forgotten, and may have contributed to the error above.